packages/xo-web/src/common/sorted-table/index.js

```diff
@@ -209,9 +209,19 @@ class IndividualAction extends Component {
       isFunction(disabled) ? disabled(item, userData) : disabled
   )
 
+  _getRedirectOnSuccess = createSelector(
+    () => this.props.item,
+    () => this.props.redirectOnSuccess,
+    () => this.props.userData,
+    (item, redirectOnSuccess, userData) =>
+      isFunction(redirectOnSuccess)
+        ? redirectOnSuccess(item, userData)
+        : redirectOnSuccess
+  )
```

Comment:
I don't understand this piece of code, what the wanted behavior here?

Make sure it's aligned with how `ActionButton`'s `redirectOnSuccess` property work :)

Reply:
To be able to generate the path from `item` and `userData`

Reply:
But `ActionButton`'s `redirectOnSuccess` can already be a function…

Reply:
`ActionButton` allows us to generate path only from the result of the handler.

Reply:
What do you mean by "that may be necessary"?
We want to be able to build the path from the `item` and `userData`. So with @Rajaa-BARHTAOUI's implementation, here's how it will be used in the `SortedTable`'s `actions`:
```js
redirectOnSuccess: (item, userData) => handlerResult =>
  myUriGenerator(item, userData, handlerResult)
```
Or like this:
```js
redirectOnSuccess: (item, userData) => myUriGenerator(item, userData)
```
Or simply:
```js
redirectOnSuccess: '/path/to/page'
```
Unfortunately **we won't be able** to use it like this:
```js
redirectOnSuccess: handlerResult => myUriGenerator(handlerResult)
```
Instead, if we only need the handler result, we'll still have to use it like this :
```js
redirectOnSuccess: () => handlerResult => myUriGenerator(handlerResult)
```
but I can't find a better way to do it. Do you?

And do you still see a potential break with this implementation?